compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1487,11 +1487,19 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 
     /// Returns the `DefId` and the `BoundRegionKind` corresponding to the given region.
-    pub fn is_suitable_region(self, mut region: Region<'tcx>) -> Option<FreeRegionInfo> {
+    pub fn is_suitable_region(
+        self,
+        mut region: Region<'tcx>,
+        generic_param_scope: LocalDefId,
+    ) -> Option<FreeRegionInfo> {
         let (suitable_region_binding_scope, bound_region) = loop {
             let def_id = match region.kind() {
```

Comment:
`let def_id = region.opt_param_def_id(generic_param_scope)?.as_local()?`

instead of manually matching on `ReLateParam` and `ReEarlyParam` here

Reply:
Oh good point, I introduced `opt_param_def_id` _after_ doing all the diagnostic changes so totally missed this.